tests/lib_testslide.py

```diff
@@ -334,3 +334,14 @@ def fails_for_wrong_type(self):
     @context.example
     def fails_for_mock_with_wrong_template(self):
         self.assert_fails(StrictMock(template=int))
+
+    @context.example
+    def passes_for_valid_forward_reference(self):
+        class Foo:
+            @classmethod
+            def get_maybe_foo(cls) -> Optional["Foo"]:  # noqa: F821 undefined name Foo
```

Comment:
Do we still need this noqa if this whole class is moved to the module level?

Reply:
Nope, though personally I prefer having it in the test that uses it (specially if no other test uses it), I'll move out though, no strong opinion.